diff --git a/src/plugins/clangformat/clangformatbaseindenter.h b/src/plugins/clangformat/clangformatbaseindenter.h
index 00dc3e9..2c1cf48 100644
--- a/src/plugins/clangformat/clangformatbaseindenter.h
+++ b/src/plugins/clangformat/clangformatbaseindenter.h
@@ -27,7 +27,7 @@
 
 #include <texteditor/indenter.h>
 
-#include <clang/Format/Format.h>
+#include <clang/Format/FormatIDE.h>
 
 namespace ClangFormat {
 
diff --git a/src/plugins/clangformat/clangformatconfigwidget.cpp b/src/plugins/clangformat/clangformatconfigwidget.cpp
index f1fa2c8..b2e552e 100644
--- a/src/plugins/clangformat/clangformatconfigwidget.cpp
+++ b/src/plugins/clangformat/clangformatconfigwidget.cpp
@@ -32,7 +32,7 @@
 #include "ui_clangformatchecks.h"
 #include "ui_clangformatconfigwidget.h"
 
-#include <clang/Format/Format.h>
+#include <clang/Format/FormatIDE.h>
 
 #include <coreplugin/icore.h>
 #include <cppeditor/cpphighlighter.h>
diff --git a/src/plugins/clangformat/clangformatplugin.cpp b/src/plugins/clangformat/clangformatplugin.cpp
index 9b76a08..840efe1 100644
--- a/src/plugins/clangformat/clangformatplugin.cpp
+++ b/src/plugins/clangformat/clangformatplugin.cpp
@@ -56,7 +56,7 @@
 #include <texteditor/icodestylepreferences.h>
 #include <texteditor/texteditorsettings.h>
 
-#include <clang/Format/Format.h>
+#include <clang/Format/FormatIDE.h>
 
 #include <utils/algorithm.h>
 
diff --git a/src/plugins/clangformat/clangformatutils.h b/src/plugins/clangformat/clangformatutils.h
index 74fd84c..7769b6a 100644
--- a/src/plugins/clangformat/clangformatutils.h
+++ b/src/plugins/clangformat/clangformatutils.h
@@ -28,7 +28,7 @@
 #include <utils/fileutils.h>
 #include <utils/id.h>
 
-#include <clang/Format/Format.h>
+#include <utils/fileutils.h>
 
 #include <QFile>
 
