diff --git a/src/Makefile b/src/Makefile
index 7ae8218..467c20a 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -23,14 +23,14 @@ export ZLIB
 CXX?=$(CROSS_COMPILE)c++
 INCLUDES=-I./core/
 DEFINES=-DPREFIX=\"$(PREFIX)\" -DSBINDIR=\"$(SBINDIR)\" -DMANDIR=\"$(MANDIR)\" -DDATADIR=\"$(DATADIR)\" -DVERSION=\"$(VERSION)\"
-CXXFLAGS=-g -Wall -g $(INCLUDES) $(DEFINES) $(RPM_OPT_FLAGS)
+CXXFLAGS += -Wall $(INCLUDES) $(DEFINES) $(RPM_OPT_FLAGS)
 ifeq ($(SQLITE), 1)
 	CXXFLAGS+= -DSQLITE $(shell pkg-config --cflags sqlite3)
 endif
 ifeq ($(ZLIB), 1)
 	CXXFLAGS+= -DZLIB $(shell pkg-config --cflags zlib)
 endif
-LDFLAGS+=-L./core/ -g
+LDFLAGS+=-L./core/
 ifneq ($(shell $(LD) --help 2| grep -- --as-needed), )
 	LDFLAGS+= -Wl,--as-needed
 endif
@@ -53,7 +53,7 @@ else
 DATAFILES = pci.ids usb.ids oui.txt manuf.txt pnp.ids pnpid.txt
 endif
 
-all: $(PACKAGENAME) $(PACKAGENAME).1 $(DATAFILES)
+all: $(PACKAGENAME) $(PACKAGENAME).1
 
 .cc.o:
 	$(CXX) $(CXXFLAGS) -c $< -o $@
@@ -63,18 +63,18 @@ all: $(PACKAGENAME) $(PACKAGENAME).1 $(DATAFILES)
 
 .PHONY: core
 core:
-	+make -C core all
+	+$(MAKE) -C core all
 
 $(PACKAGENAME): core $(PACKAGENAME).o
 	$(CXX) $(LDFLAGS) -o $@ $(PACKAGENAME).o $(LIBS)
 
 .PHONY: po
 po:
-	+make -C po all
+	+$(MAKE) -C po all
 
 .PHONY: gui
 gui: core
-	+make -C gui all
+	+$(MAKE) -C gui all
 
 .PHONY: nologo
 nologo:
@@ -85,7 +85,6 @@ static: $(PACKAGENAME)-static
 
 $(PACKAGENAME)-static: core core/lib$(PACKAGENAME).a $(PACKAGENAME).o
 	$(CXX) $(LDSTATIC) $(LDFLAGS) -o $@ $(PACKAGENAME).o $(LIBS)
-	$(STRIP) $@
 
 .PHONY: compressed
 compressed: $(PACKAGENAME)-compressed
@@ -114,14 +113,12 @@ pnp.ids:
 pnpid.txt:
 	wget http://www-pc.uni-regensburg.de/hardware/TECHNIK/PCI_PNP/pnpid.txt
 
-install: all
+install:
 	$(INSTALL) -d -m 0755 $(DESTDIR)/$(SBINDIR)
 	$(INSTALL) -m 0755 $(PACKAGENAME) $(DESTDIR)/$(SBINDIR)
 	$(INSTALL) -d -m 0755 $(DESTDIR)/$(MANDIR)/man1
 	$(INSTALL) -m 0644 $(PACKAGENAME).1 $(DESTDIR)/$(MANDIR)/man1
 	$(INSTALL) -d -m 0755 $(DESTDIR)/$(DATADIR)/$(PACKAGENAME)
-	$(INSTALL) -m 0644 $(DATAFILES) $(DESTDIR)/$(DATADIR)/$(PACKAGENAME)
-	make -C po install
 
 install-gui: gui
 	$(INSTALL) -d -m 0755 $(DESTDIR)/$(SBINDIR)
@@ -134,8 +131,8 @@ install-gui: gui
 clean:
 	rm -f $(PACKAGENAME).o $(PACKAGENAME) $(PACKAGENAME)-static $(PACKAGENAME)-compressed
 	rm -f $(addsuffix .gz,$(DATAFILES))
-	make -C core clean
-	make -C gui clean
+	$(MAKE) -C core clean
+	$(MAKE) -C gui clean
 
 depend:
 	@makedepend -Y $(SRCS) 2> /dev/null > /dev/null
diff --git a/src/core/Makefile b/src/core/Makefile
index b429380..e0dccb0 100644
--- a/src/core/Makefile
+++ b/src/core/Makefile
@@ -3,7 +3,7 @@ PACKAGENAME?=lshw
 CXX?=$(CROSS_COMPILE)c++
 INCLUDES=
 DEFINES=-DPREFIX=\"$(PREFIX)\" -DSBINDIR=\"$(SBINDIR)\" -DMANDIR=\"$(MANDIR)\" -DDATADIR=\"$(DATADIR)\"
-CXXFLAGS?=-g -Wall $(INCLUDES) $(DEFINES) $(RPM_OPT_FLAGS)
+CXXFLAGS?= -Wall $(INCLUDES) $(DEFINES) $(RPM_OPT_FLAGS)
 LDFLAGS=
 LDSTATIC=
 LIBS=
diff --git a/src/core/pci.cc b/src/core/pci.cc
index 21b9033..97d526d 100644
--- a/src/core/pci.cc
+++ b/src/core/pci.cc
@@ -20,7 +20,7 @@ __ID("@(#) $Id$");
 
 #define PROC_BUS_PCI "/proc/bus/pci"
 #define SYS_BUS_PCI "/sys/bus/pci"
-#define PCIID_PATH DATADIR"/pci.ids:/usr/share/lshw/pci.ids:/usr/local/share/pci.ids:/usr/share/pci.ids:/etc/pci.ids:/usr/share/hwdata/pci.ids:/usr/share/misc/pci.ids"
+#define PCIID_PATH "/usr/share/misc/usb.ids"
 
 #define PCI_CLASS_REVISION      0x08              /* High 24 bits are class, low 8 revision */
 #define PCI_VENDOR_ID           0x00              /* 16 bits */
diff --git a/src/gui/Makefile b/src/gui/Makefile
index f003cfb..ed75d4a 100644
--- a/src/gui/Makefile
+++ b/src/gui/Makefile
@@ -1,5 +1,7 @@
 PACKAGENAME?=lshw
 
+SQLITE?=0
+
 CXX?=$(CROSS_COMPILE)c++
 CC?=$(CROSS_COMPILE)cc
 STRIP?=strip
@@ -8,20 +10,18 @@ OBJCOPY?=objcopy
 DEFINES=-DPREFIX=\"$(PREFIX)\" -DSBINDIR=\"$(SBINDIR)\" -DMANDIR=\"$(MANDIR)\" -DDATADIR=\"$(DATADIR)\"
 GTKINCLUDES=$(shell pkg-config gtk+-2.0 --cflags)
 INCLUDES=-I../core $(GTKINCLUDES)
-CXXFLAGS=-g -Wall $(INCLUDES) $(DEFINES) $(RPM_OPT_FLAGS)
+CXXFLAGS= -Wall $(INCLUDES) $(DEFINES) $(RPM_OPT_FLAGS)
 ifeq ($(SQLITE), 1)
 	CXXFLAGS+= -DSQLITE $(shell pkg-config --cflags sqlite3)
 endif
 CFLAGS=$(CXXFLAGS) -g $(DEFINES)
 GTKLIBS=$(shell pkg-config gtk+-2.0 gmodule-2.0 --libs)
-LIBS+=-L../core -llshw -lresolv $(GTKLIBS)
+LIBS=-L../core -llshw -lresolv $(GTKLIBS)
+
 ifeq ($(SQLITE), 1)
+	CXXFLAGS+= -DSQLITE $(shell pkg-config --cflags sqlite3)
 	LIBS+= $(shell pkg-config --libs sqlite3)
 endif
-LDFLAGS=
-ifneq ($(shell $(LD) --help 2| grep -- --as-needed), )
-	LDFLAGS+= -Wl,--as-needed
-endif
 
 OBJS = gtk-lshw.o callbacks.o engine.o print-gui.o stock.o
 SRCS = $(OBJS:.o=.c)
@@ -45,8 +45,7 @@ gtk-$(PACKAGENAME): $(OBJS) ../core/liblshw.a
 	$(CXX) $(LDFLAGS) -o $@ $^ $(LIBS)
 
 install: all
-	$(STRIP) gtk-$(PACKAGENAME)
-	
+
 clean:
 	rm -f $(OBJS) gtk-$(PACKAGENAME) gtk-lshw.glade.bak gtk-lshw.gladep.bak callbacks.c.bak callbacks.h.bak Makefile.bak
 
