diff --git a/zenmap/setup.py b/zenmap/setup.py
index 6a15e87..8feda9e 100755
--- a/zenmap/setup.py
+++ b/zenmap/setup.py
@@ -186,7 +186,6 @@ class my_install(install):
         self.set_perms()
         self.set_modules_path()
         self.fix_paths()
-        self.create_uninstaller()
         self.write_installed_files()
 
     def get_installed_files(self):
@@ -204,8 +203,6 @@ class my_install(install):
                 os.path.join(self.install_data, data_dir)):
             for dir in dirs:
                 installed_files.append(os.path.join(dirpath, dir))
-        installed_files.append(
-                os.path.join(self.install_scripts, "uninstall_" + APP_NAME))
         return installed_files
 
     def create_uninstaller(self):
