diff --git a/src/cpp/r/CMakeLists.txt b/src/cpp/r/CMakeLists.txt
index 9802aaa..ec8966f 100644
--- a/src/cpp/r/CMakeLists.txt
+++ b/src/cpp/r/CMakeLists.txt
@@ -15,6 +15,18 @@
 
 project (R)
 
+# find the version of R in play
+find_package(LibR REQUIRED)
+execute_process(
+    COMMAND "${LIBR_EXECUTABLE}" "--vanilla" "--slave" "-e" "cat(as.character(getRversion()))"
+    OUTPUT_VARIABLE LIBR_VERSION)
+
+# parse and save the R version to a variable
+string(REPLACE "." ";" R_VERSION_LIST "${LIBR_VERSION}")
+list(GET R_VERSION_LIST 0 R_VERSION_MAJOR)
+list(GET R_VERSION_LIST 1 R_VERSION_MINOR)
+list(GET R_VERSION_LIST 2 R_VERSION_PATCH)
+
 # include files
 file(GLOB_RECURSE R_HEADER_FILES "*.h*")
 
diff --git a/src/cpp/r/config.h.in b/src/cpp/r/config.h.in
index 0b71e69..e60d03d 100644
--- a/src/cpp/r/config.h.in
+++ b/src/cpp/r/config.h.in
@@ -16,4 +16,10 @@
 
 #cmakedefine PANGO_CAIRO_FOUND
 
+// Important: These variables represent the version of R found during
+// compile/link time, NOT the version of R present at runtime.
+#define R_VERSION_MAJOR ${R_VERSION_MAJOR}
+#define R_VERSION_MINOR ${R_VERSION_MINOR}
+#define R_VERSION_PATCH ${R_VERSION_PATCH}
+
 
